exec/cnex/lib/string.c

```diff
@@ -0,0 +1,143 @@
+#include "string.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "array.h"
+#include "cell.h"
+#include "exec.h"
+#include "global.h"
+#include "nstring.h"
+#include "number.h"
+#include "stack.h"
+
+// ToDo: UTF8 string operations on all functions.
+
+
+void string_find(TExecutor *exec)
+{
+    TString *t = peek(exec->stack, 0)->string;
+    TString *s = peek(exec->stack, 1)->string;
+
+    int64_t ret = string_findString(s, 0, t);
+
+    pop(exec->stack);
+    pop(exec->stack);
+    push(exec->stack, cell_fromNumber(number_from_sint64(ret)));
+}
+
+void string_lower(TExecutor *exec)
+{
+    TString *ss = top(exec->stack)->string;
+
+    Cell *r = cell_newCellType(cString);
+    // Note: string_toLowerCase() creates a new string; it does not modify the existing string in-place.
+    // ToDo: Would it be better to modify the string in place?  This should be tested and benchmarked.
+    r->string = string_toLowerCase(ss);
+
+    pop(exec->stack);
+    push(exec->stack, r);
+}
+
+void string_split(TExecutor *exec)
+{
+    TString *d = peek(exec->stack, 0)->string; // Delimeter char or string
+    TString *s = peek(exec->stack, 1)->string; // Original String to split
+
+    Cell *r = cell_createArrayCell(0);
+    Cell *e = cell_newCellType(cString); // Temp element container for found strings.
+    size_t i = 0;
+    while (i < s->length) {
+        e->type = cString; // Ensure that we have the correct Cell type to operate on.
```

Comment:
Why is this necessary? Didn't you just create a new cell of type string on line 48?

Reply:
Removed / Not necessary anymore.